src/cartridges/bm_adyen/cartridge/controllers/AdyenSettings.js

```diff
@@ -0,0 +1,83 @@
+const server = require('server');
+const Transaction = require('dw/system/Transaction');
+const Logger = require('dw/system/Logger');
+const csrfProtection = require('*/cartridge/scripts/middleware/csrf');
+const AdyenConfigs = require('*/cartridge/scripts/util/adyenConfigs');
+const AdyenHelper = require('*/cartridge/scripts/util/adyenHelper');
+const constants = require('*/cartridge/adyenConstants/constants');
+
+server.get('Start', csrfProtection.generateToken, (_req, res, next) => {
+  res.render('adyenSettings/settings');
+  return next();
+});
+
+server.post('Save', server.middleware.https, (req, res, next) => {
+  try {
+    const requestBody = JSON.parse(req.body);
+    requestBody.settings.forEach((setting) => {
+      Transaction.wrap(() => {
+        AdyenConfigs.setCustomPreference(setting.key, setting.value);
+      });
+    });
+    res.json({
+      success: true,
+    });
+  } catch (error) {
+    Logger.getLogger('Adyen').error(
+      `Error while saving settings in BM configuration: ${error}`,
+    );
+    res.json({
+      success: false,
+    });
+  }
+
+  return next();
+});
+
+server.post('TestConnection', server.middleware.https, (req, res, next) => {
+  try {
+    const service = AdyenHelper.getService(
+      constants.SERVICE.CHECKOUTPAYMENTMETHODS,
+    );
+    if (!service) {
+      throw new Error('Could not do /paymentMethods call');
```

Comment:
Perhaps indicate in the error log that this is the Test Connection button and not the /paymentMethods call in the checkout. For example: "Could not verify the configured merchant account and API key"